File: Selection_Sort.c
#include<stdio.h>

void SelectionSort(int arr[], int size){
    for (int i = 0; i < size-1; i++){
        int min = i;
        for (int j = i+1; j < size; j++) if (arr[j] < arr[min]) min = j;
        
        if (min != i){
            int temp = arr[i];
            arr[i] = arr[min];
            arr[min] = temp;
        }
    }

}

int main(){
    int size;
    printf("Enter the Number of Elements: ");
    scanf("%d", &size);

    int arr[size];
    for (int i = 0; i < size; i++){
        printf("Enter the Element: ");
        scanf("%d", &arr[i]);
    }

    printf("\nBefore Sorting:\n");
    for (int i = 0; i < size; i++) printf("%d ", arr[i]);
    printf("\n");
    
    SelectionSort(arr, size);
    printf("\nAfter Sorting:\n");
    for (int i = 0; i < size; i++) printf("%d ", arr[i]);

    return 0;
}

File: Java/Question Practice Java/Selection_Sort.java
import java.util.Scanner;
public class Selection_Sort{
    public static void SelectionSort(int[] arr, int size){
        for (int i = 0; i < size-1; i++){
            int min = i;
            for (int j = i+1; j < size; j++) if (arr[j] < arr[min]) min = j;
            if (min != i){
                int temp = arr[i];
                arr[i] = arr[min];
                arr[min] = temp;
            }
        }
    }
    public static void main(String[] args){
        Scanner input = new Scanner(System.in);
        System.out.print("Enter the no of Elements: ");
        int size = input.nextInt();
        int[] arr = new int[size];

        for (int i = 0; i < size; i++){
            System.out.print("Enter the Element: ");
            arr[i] = input.nextInt();
        }

        System.out.println("\nBefore Sort:");
        for (int i = 0; i < size; i++)System.out.print(arr[i] + " ");
        
        SelectionSort(arr, size);
        System.out.println("\n\nAfter Sort:");
        for (int i = 0; i < size; i++)System.out.print(arr[i] + " ");
    }
}